### PR TITLE
Feature/new server

### PR DESCRIPTION
**Pull request by afeena**
#110 

New high-level server, compatible with aiohttp > 2.0

- Test will be fixed after merging with #136 
- `new_server` will be renamed into `server` before merging with master

### CODE REVIEW

**Comment by glaslos** [comment]
Needs a rebase.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-aiohttp<2.0
+aiohttp
```
Comment (glaslos) [inline requirements.txt]: Maybe require >=2.0 ?

File: tanner/new_server.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    @staticmethod
+    def _make_response(msg):
+        response_message = dict(
+            version=1,
+            response=dict(message=msg)
+        )
+        return response_message
+
+    @staticmethod
+    async def default_handler(request):
```
Comment (glaslos) [inline tanner/new_server.py]: This is python3.6 syntax. Did you make that clear in the README?

File: tanner/new_server.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        except (TypeError, ValueError, KeyError) as error:
+            self.logger.error('error parsing request: %s', data)
+            response_msg = self._make_response(msg=type(error).__name__)
+        else:
+            session = await self.session_manager.add_or_update_session(
+                data, self.redis_client
+            )
+            self.logger.info('Requested path %s', path)
+            await self.dorks.extract_path(path, self.redis_client)
+            detection = await self.base_handler.handle(data, session, path)
```
Comment (glaslos) [inline tanner/new_server.py]: Maybe rename `handle()` into `detect` or `classify`? Should probably go into a separate PR :)

**Comment by glaslos** [comment]
Looks good! Do you need another review or ready to merge?

File: tanner/new_server.py (hunk 3 above)
Comment (afeena) [inline tanner/new_server.py]: I like the idea to make the names more specific :)

**Comment by afeena** [comment]
@glaslos Thanks for review :) 
I want to discuss the `async/await` approach. Maybe it will be better to use `yield from` construction on this stage?

**Comment by glaslos** [comment]
I don't mind the 3.6 syntax. It might cause issues on older Linux distributions. What is your opinion?

**Comment by afeena** [comment]
We use `uvloop` which needs python 3.5, `async/await` is available since python 3.5.3. So I think we can migrate to new syntax without any big damage.

**Comment by glaslos** [comment]
Ah, you are right! Go for it!

**Comment by afeena** [comment]
Ooops, I forgot to push changes. My bad.
